### PR TITLE
Privtmpl

### PR DESCRIPTION
**Pull request by pavanb018**
Comments provided in PR 241 are addressed and the commits for those are in this PR.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@pavanb018 You will have to add license headers to the new files

**Comment by pavanb018** [comment]
test_project_private_template (integration.component.test_project_template.TestProjectPrivateTemplate) ... === TestName: test_project_private_template | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 216.074s

OK

**Comment by pavanb018** [comment]
On Basic Zone:

Test Deploy Virtual Machine ... === TestName: test_deploy_vm_with_tags | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 34.365s

OK

**Comment by pavanb018** [comment]
On Advanced Zone:

Test Deploy Virtual Machine ... === TestName: test_deploy_vm_with_tags | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 120.020s

OK

File: test/integration/component/test_project_template.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        self.virtual_machine.stop(self.api_client)
+
+        list_volumes = Volume.list(
+                                   self.api_client,
+                                   projectid=self.project.id,
+                                   virtualmachineid=self.virtual_machine.id,
+                                   type='ROOT',
+                                   listall=True
+                                   )
+
```
Comment (nitt10prashant) [inline test/integration/component/test_project_template.py]: use validate list to validate list_volumes

File: test/integration/component/test_project_template.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        self.debug("Created template with ID: %s" % project_template.id)
+
+        #api_client = cls.testClient.getUserApiClient(UserName=cls.account.name, DomainName=cls.account.domain)
+
+        list_template_response = Template.list(
+                                    self.api_client,
+                                    templatefilter=self.services["project"]["templatefilter"],
+                                    projectid=self.project.id
+                                    )
+
```
Comment (nitt10prashant) [inline test/integration/component/test_project_template.py]: do same for list_template_response

File: test/integration/component/test_user_taglisting.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        )
+
+        tags = Tag.list(
+            self.apiclient,
+            listall=True,
+            resourceType='userVM',
+            key='vmtag'+self.virtual_machine.id,
+            value='autotag'+self.virtual_machine.id
+        )
+        self.assertEqual(
```
Comment (nitt10prashant) [inline test/integration/component/test_user_taglisting.py]: use validate list

File: test/integration/component/test_project_template.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        project_template = Template.create(
+                                self.api_client,
+                                self.services["project"],
+                                self.volume.id,
+                                projectid=self.project.id
+                                )
+        self.debug("Created template with ID: %s" % project_template.id)
+
+        #api_client = cls.testClient.getUserApiClient(UserName=cls.account.name, DomainName=cls.account.domain)
```
Comment (nitt10prashant) [inline test/integration/component/test_project_template.py]: i think this line is not required  .. (#api_client = ..)

**Comment by wido** [comment]
I think it looks good? But it depends on earlier PR's I think since it's only a test

**Comment by rodrigo93** [comment]
Why does this PR still open? Shouldn't be closed? Seems that has been resolved already...

**Pull request closed without merging**